src/api/eventTypes.js

```diff
@@ -110,6 +110,13 @@ export type PresenceEvent = {|
   presence: UserPresence,
 |};
 
+export type UserStatusEvent = {|
+  ...EventCommon,
+  user_id: number,
+  away?: boolean,
+  status_text?: string,
```

Comment:
Huh interesting! Are both of these really optional?

Also: Is your description here based on empirical observation, or reading the server code, or discussions with Steve, or something else?

Reply:
Based on all three. One or two of them exist. Never none (as expected).

Reply:
Ah cool, I see that the `UserStatus` types in the app's own data structures now has some quite helpful jsdoc, and of course that overlaps with this information.

No need to duplicate info. Any given piece of information can be written down in one of the places where it's important, and then write "See also `UserStatus` in src/types.js" or whatever in each of the other places.

I think the key bit of info that that doesn't cover and is important for this event type is: what does it mean when one of these is missing? Also `away: false`, which can't occur in the other place.

(Ideally I think the information on that `UserStatus` type would be in `api/` somewhere -- particularly as we seem to use it in `InitialData`, which is really mainly a matter of describing the API. But it's fine to defer doing that rearrangement.)

Reply:
Indeed. I had a commit somewhere that I think I dropped because it was causing too many merge conflicts that moved InitialData and all related types to `apiTypes`.